extensions/interactions/baseValidatorSpec.js

```diff
@@ -30,12 +30,12 @@ describe('Interaction validator', function() {
   var currentState, otherState, goodOutcomeDest, goodOutcomeFeedback;
   var badOutcome, goodAnswerGroups, goodDefaultOutcome;
 
-  var createAnswerGroup = function(outcome, ruleSpecs, correct) {
-    if (!ruleSpecs) {
-      ruleSpecs = [];
+  var createAnswerGroup = function(outcome, rules, correct) {
+    if (!rules) {
+      rules = [];
     }
     return {
-      ruleSpecs: ruleSpecs,
+      rules: rules,
       outcome: outcome,
       correct: !!correct
     };
```

Comment:
I think it makes sense to use the factory. The !! is just to convert 'correct' into a boolean. I don't think it's needed, actually, but please check that every call to createAnswerGroup contains exactly 3 args.